src/Adapter.ts

```diff
@@ -60,6 +75,23 @@ export default class Adapter extends EnzymeAdapter {
     // Work around a bug in Enzyme where `ShallowWrapper.getElements` calls
     // the `nodeToElement` method with undefined `this`.
     this.nodeToElement = this.nodeToElement.bind(this);
+
+    if (this.preactAdapterOptions.useCompatShallowRendering) {
+      this.isFragment = node => node?.type === Fragment;
+
+      this.displayNameOfNode = (node: RSTNode | null): string | null => {
```

Comment:
By implementing this method, I no longer need to set `Fragment.displayName`. Yay no side-effects! This method is basically [the default implementation Enzyme provides](https://github.com/enzymejs/enzyme/blob/9bb92eb6dad19152766549bb22039264982e673b/packages/enzyme/src/Debug.js#L18-L26) plus the special Fragment handling.